server/src/main/java/org/elasticsearch/search/aggregations/metrics/TopHitsAggregator.java

```diff
@@ -198,7 +199,7 @@ public InternalAggregation buildAggregation(long owningBucketOrdinal) throws IOE
         for (int i = 0; i < topDocs.scoreDocs.length; i++) {
             docIdsToLoad[i] = topDocs.scoreDocs[i].doc;
         }
-        FetchSearchResult fetchResult = runFetchPhase(subSearchContext, docIdsToLoad);
+        FetchSearchResult fetchResult = runFetchPhase(subSearchContext, docIdsToLoad, this::addRequestCircuitBreakerBytes);
```

Comment:
Should we batch here and avoid invoking the CB for every document? 
Maybe `addRequestCircuitBreakerBytes` should be responsible for this?

I suspect that fetching source is way more expensive than invoking the CB, so I'm not sure we want more complication here.

Reply:
I ran nyc-taxis track, and I see no slowdown (actually it seems faster).
Now I'm running NOAA, that contains more aggs